iroha_data_model/src/isi.rs

```diff
@@ -277,6 +359,50 @@ where
     }
 }
 
+#[allow(clippy::len_without_is_empty)]
+impl SetKeyValueBox {
+    /// Calculates number of underneath instructions and expressions
+    pub fn len(&self) -> usize {
+        self.object_id.len() + self.key.len() + self.value.len() + 1
+    }
+
+    /// Default [`SetKeyValueBox`] constructor.
+    pub fn new<
+        I: Into<EvaluatesTo<IdBox>>,
+        K: Into<EvaluatesTo<String>>,
+        V: Into<EvaluatesTo<Value>>,
+    >(
+        object_id: I,
+        key: K,
+        value: V,
+    ) -> Self {
```

Comment:
What do you think about impl form of this?
```rust
pub fn new(
        object_id: impl Into<EvaluatesTo<IdBox>>,
        key: impl Into<EvaluatesTo<String>>,
        value: impl Into<EvaluatesTo<Value>>,
    ) -> Self;
```

Reply:
This is an internal structure it is constructed only after everything is already evaluated, so yes there is a similar `new` and structure with expressions they have the `Box` prefix.

Reply:
The structure with generics just serve us to better subdivide ISI execution when we branch on enum variants

Reply:
Yeah, I know. But I mean just to make this form shorter, clearer and easier to read, while remaining same.

Reply:
I mean there should not be any evaluation at that stage. `EvaluatesTo` wraps an expression, and there should not be expressions already, everything is raw value there.

So the implementation of this structure will be something like that:
`impl Execute for SetKeyValue<Asset, String, Value> {`

This structure has bounds for `ValueMarker` which means it is a raw, already evaluated value.

Reply:
We discussed it with Ivan on a zoom call. The question was of whether to use `impl ... ` or not in trait bounds. We decided to leave it as it is for now and think about adding a rule for it to the style guide.